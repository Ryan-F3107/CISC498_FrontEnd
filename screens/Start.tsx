import React from 'react';
import { Text, View, Button } from 'react-native';
import styles from '../styling/styles';
function Start({ navigation }) {
  return (
    <View style={styles.startContainer}>
      <Text>First page of app -- App.tsx</Text>
      <Button
        title="Login"
      //onPress = {()=> Alert.alert("Button pressed")} 
      />
      <Button
        title="Sign Up"
        onPress={() => navigation.navigate('SignUpDefault')}
      />
    </View>
  );
}

export default Start;
