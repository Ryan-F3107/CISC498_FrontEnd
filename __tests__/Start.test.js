import React from 'react';
import renderer from 'react-test-renderer';
import Start from '../screens/Start';

describe("<Start />", () => {
    it("has 3 children", () => {
        const tree = renderer.create(<Start />).toJSON();
        expect(tree.children.length).toBe(3);
    });

    it('renders correctly', () => {
        renderer.create(<Start />);
    });

    it('Start testing with snapshot', () => {
        const tree = renderer.create(<Start />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});