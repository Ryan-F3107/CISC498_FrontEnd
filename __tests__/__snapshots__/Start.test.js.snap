// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Start /> Start testing with snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e8efff",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 250,
        "marginBottom": 100,
        "marginTop": 100,
        "resizeMode": "contain",
        "width": 250,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0a0540",
        "borderColor": "#0a0540",
        "borderRadius": 30,
        "borderWidth": 1.5,
        "opacity": 1,
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "padding": 10,
          "paddingLeft": 60,
          "paddingRight": 60,
        }
      }
    >
      Sign In
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#0a0540",
          "borderRadius": 30,
          "borderWidth": 1.5,
          "opacity": 1,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0a0540",
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        Create an Account
      </Text>
    </View>
  </View>
</View>
`;
